drivers/sensor/adi/ad2s1210/ad2s1210.c

```diff
@@ -0,0 +1,904 @@
+
+	ret = ad2s1210_reg_write(dev, AD2S1210_REG_CONTROL, control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (!data->have_resolution_pins) {
+		data->resolution = resolution;
+		return 0;
+	}
+
+	/* Set RES0 pin */
+	ret = gpio_pin_set_dt(&config->resolution_gpios[AD2S1210_RES_PIN_RES0],
+			      !!(control & AD2S1210_CONTROL_RES0_MASK));
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Set RES1 pin */
+	ret = gpio_pin_set_dt(&config->resolution_gpios[AD2S1210_RES_PIN_RES1],
+			      !!(control & AD2S1210_CONTROL_RES1_MASK));
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->resolution = resolution;
+	return 0;
+}
+
+static int ad2s1210_set_hysteresis(const struct device *dev, bool enable)
+{
+	int ret;
+	uint8_t control;
+
+	ret = ad2s1210_reg_read(dev, AD2S1210_REG_CONTROL, &control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	control &= ~AD2S1210_ENABLE_HYSTERESIS;
+	if (enable) {
+		control |= AD2S1210_ENABLE_HYSTERESIS;
+	}
+
+	return ad2s1210_reg_write(dev, AD2S1210_REG_CONTROL, control);
+}
+
+static __maybe_unused int
+ad2s1210_hysteresis_is_enabled(const struct device *dev) /* cppcheck-suppress unusedFunction */
+{
+	int ret;
+	uint8_t control;
+
+	ret = ad2s1210_reg_read(dev, AD2S1210_REG_CONTROL, &control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (control & AD2S1210_ENABLE_HYSTERESIS) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static __maybe_unused int
+ad2s1210_reinit_excitation_frequency(const struct device *dev,
+				     uint16_t fexcit) /* cppcheck-suppress unusedFunction */
+{
+	const struct ad2s1210_config *config = dev->config;
+	int ret;
+	uint32_t fcw;
+
+	/* Calculate the frequency control word */
+	fcw = (fexcit * (1 << 15)) / config->clock_frequency;
+	if (fcw < AD2S1210_MIN_FCW || fcw > AD2S1210_MAX_FCW) {
+		return -ERANGE;
+	}
+
+	/* Write the frequency control word */
+	ret = ad2s1210_reg_write(dev, AD2S1210_REG_EXCIT_FREQ, (int8_t)fcw);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Software reset to reinitialize excitation frequency output */
+	return ad2s1210_reg_write(dev, AD2S1210_REG_SOFT_RESET, 0);
+}
+
+static int ad2s1210_init(const struct device *dev) /* cppcheck-suppress unusedFunction */
+{
+	const struct ad2s1210_config *config = dev->config;
+	struct ad2s1210_data *data = dev->data;
+	int ret;
+	enum ad2s1210_res resolution;
+
+	/* Check if SPI bus is ready */
+	if (!spi_is_ready_dt(&config->spi)) {
+		LOG_ERR("SPI bus not ready");
+		return -ENODEV;
+	}
+
+	/* Check if sample GPIO port is ready */
+	if (!device_is_ready(config->sample_gpio.port)) {
+		LOG_ERR("Sample GPIO port not ready");
+		return -ENODEV;
+	}
+
+	/* Check if mode GPIO ports are ready (required) */
+	for (uint8_t idx = 0; idx < (uint8_t)AD2S1210_MODE_PIN_MAX_VAL; idx++) {
+		if (!device_is_ready(config->mode_gpios[idx].port)) {
+			LOG_ERR("Mode GPIO %d port not ready", idx);
+			return -ENODEV;
+		}
+	}
+
+	/* Check if resolution GPIO ports are ready */
+
+	if (config->resolution_gpios[0].port != NULL || config->resolution_gpios[1].port != NULL) {
```

Comment:
If `data->have_resolution_pins` is set in init this can be
```suggestion
	if (config->have_resolution_pins) {
```

Reply:
I'm moving the flag.